mindmeld/augmentation.py

```diff
@@ -0,0 +1,345 @@
+                "Library not found: 'torch'. Run 'pip install mindmeld[augment]' to install."
+            )
+
+        if not _is_module_available("transformers"):
+            raise ModuleNotFoundError(
+                "Library not found: 'transformers'. Run 'pip install mindmeld[augment]' to install."
+            )
+
+    def _check_lang_support(self):
+        """Checks if language is currently supported for augmentation."""
+        if self.lang not in SUPPORTED_LANG_CODES:
+            raise UnsupportedLanguageError(
+                f"'{self.lang}' is not supported yet. "
+                "English (en), French (fr), and Italian (it), Portuguese (pt), Romanian (ro) "
+                " and Spanish (es) are currently supported."
+            )
+
+    def augment(self, **kwargs):
+        """Augments queries given initial queries in application."""
+        filtered_paths = self._get_files(paths=self.paths)
+
+        for path in tqdm(filtered_paths):
+            queries = self._read_path_queries(path)
+            augmented_queries = self.augment_queries(queries, **kwargs)
+            self._write_files(path, augmented_queries, suffix=self.path_suffix)
+
+    @abstractmethod
+    def augment_queries(self, queries):
+        """Generates augmented data given application queries.
+
+        Args:
+            queries (list): List of queries.
+
+        Return:
+            augmented_queries (list): List of augmented queries.
+        """
+        raise NotImplementedError("Subclasses must implement this method")
+
+    def _get_files(self, paths=None):
+        """Fetches relevant files given the path rules specified in the config.
+
+        Args:
+            paths (list): Path rules for fetching relevant files.
+
+        Return:
+            filtered_paths (list): List of file paths to be augmeted.
+        """
+        all_file_paths = self._resource_loader.get_all_file_paths()
+
+        if not paths:
+            logger.warning(
+                """'paths' field is not configured or misconfigured in the `config.py`.
+                 Can't find files to augment."""
+            )
+            return []
+
+        rules = paths
+
+        filtered_paths = []
+
+        for rule in rules:
+            pattern = get_pattern(rule)
+            compiled_pattern = re.compile(pattern)
+            filtered_paths.extend(
+                self._resource_loader.filter_file_paths(
+                    compiled_pattern=compiled_pattern, file_paths=all_file_paths
+                )
+            )
+        return filtered_paths
+
+    def _read_path_queries(self, path):
+        """Returns all queries in a specified file path.
+
+        Args:
+            path (str): File path.
+
+        Return:
+            queries (list): List of queries in the file path.
+        """
+        with open(path, "r") as f:
+            queries = f.readlines()
+        return queries
+
+    def _write_files(self, path, augmented_queries, suffix):
```

Comment:
I think `_write_to_file` seems more descriptive?

Reply:
I also think these functions seem like utility functions that dont need to be a member of this class. The prefix can also be generalized